kube-monitoring/charts/templates/_datasource.tpl

```diff
@@ -0,0 +1,70 @@
+# Configuration file version
+apiVersion: 1
+
+# List of data sources to delete from the database.
+deleteDatasources:
+#  - name: data-source-name
+#    orgId: 1
+
+# List of data sources to insert/update depending on what's
+# available in the database.
+datasources:
+  # <string, required> Sets the name you use to refer to
+  # the data source in panels and queries.
+  - name: {{ printf "%s-%s" $.Release.Name "prometheus" }}
+    # <string, required> Sets the data source type.
+    type: prometheus
+    # <string, required> Sets the access mode, either
+    # proxy or direct (Server or Browser in the UI).
+    # Some data sources are incompatible with any setting
+    # but proxy (Server).
+    access: proxy
+    # <int> Sets the organization id. Defaults to orgId 1.
+    orgId: 1
+    # <string> Sets a custom UID to reference this
+    # data source in other parts of the configuration.
+    # If not specified, Plutono generates one.
+    uid:
+    # <string> Sets the data source's URL, including the
+    # port.
+    url: {{ printf "http://%s-prometheus.%s.svc:9090" $.Release.Name $.Release.Namespace }}
+    # <string> Sets the database user, if necessary.
+    user:
+    # <string> Sets the database name, if necessary.
+    database:
+    # <bool> Enables basic authorization.
+    basicAuth:
+    # <string> Sets the basic authorization username.
+    basicAuthUser:
+    # <bool> Enables credential headers.
+    withCredentials:
```

Comment:
since they are not templated they always will be empty. do we need them or can we comment them out?

Reply:
You're right, this could be a template. But this is the first iteration and I would leave it open as a feature moving forward.